app/Helpers/database/player-game.php

```diff
@@ -276,20 +277,33 @@ function reactivateUserEventAchievements(User $user, array $userUnlocks): array
 
 function GetAllUserProgress(User $user, int $consoleID): array
 {
-    $retVal = [];
-
-    $query = "SELECT gd.ID, gd.achievements_published AS NumAch,
-                     COALESCE(pg.achievements_unlocked, 0) AS Earned,
-                     COALESCE(pg.achievements_unlocked_hardcore, 0) AS HCEarned
-            FROM GameData AS gd
-            LEFT JOIN player_games pg ON pg.game_id = gd.ID AND pg.user_id={$user->id}
-            WHERE gd.achievements_published > 0 AND gd.ConsoleID = $consoleID";
+    /** @var Collection<int, PlayerGame> $playerGames */
+    $playerGames = $user->playerGames()
+        ->whereIn('game_id', function ($query) use ($consoleID) {
+            $query->select('ID')
+                  ->from('GameData')
+                  ->where('ConsoleID', $consoleID)
+                  ->where('achievements_published', '>', 0);
+        })
+        ->get()
+        ->keyBy('game_id');
 
-    foreach (legacyDbFetchAll($query) as $row) {
-        $id = $row['ID'];
-        unset($row['ID']);
+    /** @var Collection<int, Game> $games */
+    $games = Game::where('ConsoleID', $consoleID)
+        ->where('achievements_published', '>', 0)
+        ->get();
 
-        $retVal[$id] = $row;
+    /** @var array<int, array{NumAch: int, Earned: int, HCEarned: int}> $retVal */
+    $retVal = [];
+    foreach ($games as $game) {
+        /** @var ?PlayerGame $playerGame */
+        $playerGame = $playerGames->get($game->id);
+
+        $retVal[$game->id] = [
+            'NumAch' => $game->achievements_published,
+            'Earned' => $playerGame?->achievements_unlocked ?? 0,
+            'HCEarned' => $playerGame?->achievements_unlocked_hardcore ?? 0,
```

Comment:
Since this API has been broken for an indeterminate amount of time, should we take the opportunity to rename these fields?

Reply:
I think so. Maybe what would be best is to not alias them at all:
```json
'AchievementsPublished' => ...,
'AchievementsUnlocked' => ...,
'AchievementsUnlockedHardcore' => ...,
```

Reply:
That might be too verbose. Remember that this API returns entries for every game on the system and several of our systems have more than 2000 entries.

Maybe `Achievements`, `Unlocked`, and `UnlockedHardcore`?

Reply:
To be honest, I can't help but feel that if we're going to bring this endpoint back after it has been offline for so long, the current response shape seems kinda undesirable.

For example, if I set `c` to 12, I get 723 objects returned in the response. It may be better to just exclude values where the user has no progress. I'm going to optimistically assume this is okay and do this in latest. If you're against the change, I will roll it back.

Alternatively, if we actually do need all the game IDs in the response, maybe we can cut "Unlocked" and "UnlockedHardcore" if those values are 0.

Reply:
> I'm going to optimistically assume this is okay and do this in latest.

I'm not sure exactly what the requirements are, but it seems like filtering is undesirable.
https://discord.com/channels/476211979464343552/757767535293890682/1321014964453904419

Reply:
Gotcha. I've fallen back to this strategy in latest:
> Alternatively, if we actually do need all the game IDs in the response, maybe we can cut "Unlocked" and "UnlockedHardcore" if those values are 0.